libyul/AsmAnalysis.cpp

```diff
@@ -353,14 +349,16 @@ void AsmAnalyzer::operator()(Switch const& _switch)
 	{
 		if (_case.value)
 		{
+			auto watcher = m_errorReporter.errorWatcher();
+
 			expectType(valueType, _case.value->type, _case.value->location);
 
 			// We cannot use "expectExpression" here because *_case.value is not an
 			// Expression and would be converted to an Expression otherwise.
 			(*this)(*_case.value);
 
 			/// Note: the parser ensures there is only one default case
-			if (m_success && !cases.insert(valueOfLiteral(*_case.value)).second)
+			if (watcher() && !cases.insert(valueOfLiteral(*_case.value)).second)
 				declarationError(_case.location, "Duplicate case defined.");
```

Comment:
I think this changes behavior:
- Before your change this function was not resetting `m_success` and `(*this)(*_case.value)` is `operator()(Literal)` which was never touching it either. This means that you would get `"Duplicate case defined."` error only if nothing else set `m_success` to `false` before visiting the switch and you would only get one error like this per switch.
- In your version the watcher gets reset for each case so you will get the message for every duplicate case.

Now, I'm not sure which one is actually desirable. Personally, I think it's fine to report all the duplicates but it might have been was intentionally designed like this.

Reply:
Yes, indeed. I'm aware that there are minor changes in behavior, as we are switching from the class-level `m_success` flag to the function-level `watcher`s. I hope these changes are safe (and for better), but would be good to get a confirmation from a code owner.